csharp/src/Ice/OutputStream.cs

```diff
@@ -1629,7 +1648,11 @@ internal Position Save()
         internal Position WriteEmptyEncapsulation(Encoding encoding)
         {
             encoding.CheckSupported();
-            WriteEncapsulationHeader(size: 2, encoding, sizeLength: 1);
+            WriteEncapsulationHeader(size: encoding == Encoding.V2_0 ? 3 : 2, encoding, sizeLength: 1);
+            if (encoding == Encoding.V2_0)
+            {
+                WriteByte(0); // Placeholder for the compression status
```

Comment:
You mean it's the compression status? Placeholder implies it will be rewritten later.

Reply:
right is the compression  status it will be rewrite if compressed, but it is not a placeholder